.github/workflows/e2e-tests.yml

```diff
@@ -0,0 +1,114 @@
+name: E2E Tests
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, labeled]
+
+  push:
+    branches:
+      - 'main'
+      - 'v*'
+
+env: 
+  GITHUB_USER_TOKEN: "${{ secrets.MESOSPHERECI_USER_TOKEN }}"
+jobs:
+  ensure-ci:
+    name: Ensure CI Image
+    if: ${{ github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'ok-to-test') }}
+    runs-on:
+      - self-hosted
+      - small
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
+
+      - name: Install asdf & tools
+        uses: asdf-vm/actions/install@v2
+
+      - name: make CI image
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+        run: make GOOS=linux GOARCH=amd64 ci.docker.push
+
+  install-e2e-tests:
+    name: Install E2E Tests
+    needs: [ensure-ci]
+    if: ${{ github.event_name != 'pull_request' || contains(github.event.pull_request.labels.*.name, 'ok-to-test') }}
+    runs-on:
+      - self-hosted
+      - kind-large
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
+
+      - name: Setup go
```

Comment:
it looks like the upgrade tests below don't have this step and they passed, maybe we can drop this?